### PR TITLE
8342884: RISC-V: verify float <--> float16 conversion

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you help to review this simple patch? it removes the `Experimental` of the `UseZvfh`.
Thanks

Currently, only float <--> float16 conversions use Zvfh extension, I've run the jmh tests on bananapi, the performance result shows it's good.
<google-sheets-html-origin style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">
Benchmark-XX:+UseZfh -XX:+UnlockExperimentalVMOptions -XX:+/-UseZvfh | (size) | Mode | Cnt | Score -intrinsic | Score +intrinsic | Error | Units | Improvement
-- | -- | -- | -- | -- | -- | -- | -- | --
Fp16ConversionBenchmark.float16ToFloat | 2048 | avgt | 10 | 8129.72 | 4729.125 | 71.937 | ns/op | 1.719
Fp16ConversionBenchmark.float16ToFloatMemory | 2048 | avgt | 10 | 16.9 | 16.894 | 0.002 | ns/op | 1
Fp16ConversionBenchmark.floatToFloat16 | 2048 | avgt | 10 | 12561.962 | 3767.944 | 12.652 | ns/op | 3.334
Fp16ConversionBenchmark.floatToFloat16Memory | 2048 | avgt | 10 | 18.146 | 18.147 | 0.003 | ns/op | 1

</google-sheets-html-origin>

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342884](https://bugs.openjdk.org/browse/JDK-8342884): RISC-V: verify float &lt;--&gt; float16 conversion (**Sub-task** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21664/head:pull/21664` \
`$ git checkout pull/21664`

Update a local copy of the PR: \
`$ git checkout pull/21664` \
`$ git pull https://git.openjdk.org/jdk.git pull/21664/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21664`

View PR using the GUI difftool: \
`$ git pr show -t 21664`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21664.diff">https://git.openjdk.org/jdk/pull/21664.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21664#issuecomment-2432192063)

### CODE REVIEW

File: src/hotspot/cpu/riscv/globals_riscv.hpp [hunk 1]
```diff
@@ -116,7 +116,7 @@ define_pd_global(intx, InlineSmallCode,          1000);
   product(bool, UseZihintpause, false, EXPERIMENTAL,                             \
           "Use Zihintpause instructions")                                        \
   product(bool, UseZvbb, false, EXPERIMENTAL, "Use Zvbb instructions")           \
-  product(bool, UseZvfh, false, EXPERIMENTAL, "Use Zvfh instructions")           \
+  product(bool, UseZvfh, false, "Use Zvfh instructions")                         \
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/globals_riscv.hpp:119]: Nit: Better to move it out of the `EXPERIMENTAL` group. Maybe immediately after the line for `UseZfh`.

**Review by RealFYang — CHANGES_REQUESTED** [review]
Thanks for carrying the verification.

File: src/hotspot/cpu/riscv/globals_riscv.hpp (hunk 1 above)
Comment (Hamlin-Li) [reply src/hotspot/cpu/riscv/globals_riscv.hpp:119]: > Nit: Better to move it out of the `EXPERIMENTAL` group. Maybe immediately after the line for `UseZfh`.
> 

Currently the pattern is ordering by name, so I'd prefer to keep it. :)

> (Seems we should also auto-enable this feature through hwprobe like other non-experimental flags. I see macros `RISCV_HWPROBE_EXT_ZVFH` and `RISCV_HWPROBE_EXT_ZVFHMIN` in file os_cpu/linux_riscv/riscv_hwprobe.cpp, but neither of them are checked in function `RiscvHwprobe::add_features_from_query_result`)

Thanks, add test of RISCV_HWPROBE_EXT_ZVFH, but not RISCV_HWPROBE_EXT_ZVFHMIN, as we don't have the corresponding define and usage ext_Zvfhmin and so on.

**Review by RealFYang — APPROVED** [review]
Latest version looks good. Thanks.

**Review by luhenry — APPROVED** [review]

**Pull request closed without merging**